prog/dftb+/lib_dftbplus/initprogram.F90

```diff
@@ -1694,7 +1734,15 @@ subroutine initProgramVariables(input, env)
     tForces = input%ctrl%tForces .or. tPrintForces
     tLinResp = input%ctrl%lrespini%tInit
 
-    referenceN0(:,:) = input%slako%skOcc(1:orb%mShell, :)
+    select case(hamiltonianType)
+    case default
+      call error("Invalid Hamiltonian")
+    case(hamiltonianTypes%dftb)
+      referenceN0(:,:) = input%slako%skOcc(1:orb%mShell, :)
```

Comment:
Here `input%slako%skOcc` is copied to `referenceN0`, might need another copy, if  `referenceN0` is modified.